lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -172,6 +175,43 @@ def add_files_to_build_phases
             end
           end
 
+          # Adds the test app host targets for the library to the Pods project with the
+          # appropriate build configurations.
+          #
+          # @return [void]
+          #
+          def add_test_app_host_targets
+            target.test_specs.each do |test_spec|
+              next unless test_spec.consumer(target.platform).requires_app_host?
+              name = target.app_host_label(test_spec.test_type)
+              platform_name = target.platform.name
+              app_host_target = !project.targets.find { |t| t.name == name }
+              unless app_host_target.nil?
+                app_host_target = Pod::Generator::AppTargetHelper.add_app_target(project, platform_name, deployment_target, name)
+                app_host_target.build_configurations.each do |configuration|
+                  configuration.build_settings.merge!(custom_build_settings)
+                  configuration.build_settings['PRODUCT_NAME'] = name
+                  configuration.build_settings['PRODUCT_BUNDLE_IDENTIFIER'] = 'org.cocoapods.${PRODUCT_NAME:rfc1034identifier}'
+                end
+                Pod::Generator::AppTargetHelper.add_app_host_main_file(project, app_host_target, platform_name, name)
+                app_host_info_plist_path = project.path.dirname.+("#{name}/Info.plist")
+                create_info_plist_file(app_host_info_plist_path, app_host_target, '1.0.0', target.platform, :appl)
+              end
+              # Wire all test native targets with the app host.
+              native_test_target = target.native_target_for_spec(test_spec)
+              native_test_target.build_configurations.each do |configuration|
+                test_host = "$(BUILT_PRODUCTS_DIR)/#{name}.app/"
+                test_host << 'Contents/MacOS/' if target.platform == :osx
+                test_host << name.to_s
+                configuration.build_settings['TEST_HOST'] = test_host
+              end
+              # TODO THIS IS UGLY.
+              root_attributes = project.root_object.attributes['TargetAttributes'] || {}
```

Comment:
@segiddins it appears that when wiring up the app host if I don't do this Xcode will mark my app host as "Custom". Granted this is ugly do you have any ideas/thoughts perhaps to hide this within Xcodeproj?